curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/633928c6b0baf16c0b0d823b.md

```diff
@@ -0,0 +1,96 @@
+---
+id: 633928c6b0baf16c0b0d823b
+title: Step 41
+challengeType: 0
+dashedName: step-41
+---
+
+# --description--
+
+You have created two radio buttons. However, if you submit the form, it does not require you to check any of them.
+
+This is not good, since you need to know what type of account the user requires.
+
+To correct this add a `required` attribute to `personal-account or business-account`.
+
+# Code Explanation
+
+Since our `radio buttons` are linked together due to the `name` attribute, we can give the `required` field to any of the radio buttons, and it will apply to the entire group.
+
+# --hints--
+
+You should use the `required` attribute for the `input` tag.
+
+```js
+assert.equal(document.getElementsByName("account-type").required, true);
+```
+
+# --seed--
+
+## --seed-contents--
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8">
+    <title>Registration Form</title>
+    <link rel="stylesheet" href="styles.css" />
+  </head>
+  <body>
+    <h1>Registration Form</h1>
+    <p>Please fill out this form with the required information</p>
+    <form action='https://register-demo.freecodecamp.org'>
+      <fieldset>
+        <label for="first-name">Enter Your First Name: <input id="first-name" name="first-name" type="text" required /></label>
+        <label for="last-name">Enter Your Last Name: <input id="last-name" name="last-name" type="text" required /></label>
+        <label for="email">Enter Your Email: <input id="email" name="email" type="email" required /></label>
+        <label for="new-password">Create a New Password: <input id="new-password" name="new-password" type="password" pattern="[a-z0-5]{8,}" required /></label>
+      </fieldset>
+      <fieldset>
+        <label for="personal-account"><input id="personal-account" type="radio" name="account-type" /> Personal</label>
+        <label for="business-account"><input id="business-account" type="radio" name="account-type" /> Business</label>
```

Comment:
You never added the `legend`, so now it doesn't make sense to have `Personal` and `Business` on their own there